src/gvars.c

```diff
@@ -307,6 +307,20 @@ static Obj * ELM_COPS_PLIST(Obj cops, UInt i)
 void            AssGVar (
     UInt                gvar,
     Obj                 val )
+{
+    /* make certain that the variable is not read only                     */
+    if ((REREADING != True) &&
+        (ELM_GVAR_LIST(WriteGVars, gvar) == INTOBJ_INT(0))) {
+        ErrorMayQuit("Variable: '%g' is read only", (Int)NameGVar(gvar), 0);
+    }
+
+    AssGVar_NoReadOnlyCheck(gvar, val);
+}
+
+// This is a kernel-only variant of AssGVar which will change read-only
+// variables, which is used for constants like:
+// Time, MemoryAllocated, last, last2, last3
+void AssGVar_NoReadOnlyCheck(UInt gvar, Obj val)
```

Comment:
Maybe `ForceAssGVar`?

Reply:
The problem with force is this removes the 'ReadOnly' check, but not the 'Constant' check (because we use that to improve parsing, and we really can't change those, ever).

Reply:
My concern about `AssGVar_NoReadOnlyCheck` is that the underscore breaks our naming conventions. So I'd prefer `AssGVarNoReadOnlyCheck` or `AssGVarIgnoreReadOnly` or whatever. I also still think `ForceAssGVar` is good, as to me at least it is kinda clear that it only "forces" / ignores "optional" restrictions, but won't violate an important system invariant like constness. But of course that's just my view, if others (you) find this misleading, then we should avoid it.

Reply:
I've gone for `AssGVarWithoutReadOnlyCheck`. My main problem with `Force` is just I could imagine a bunch of things I could force.